### PR TITLE
Added SpringBoot3-GraphQL sample

### PR DESCRIPTION
**Pull request by mbfreder**
*Issue #715:*

*Description of changes:*

Added SpringBoot3-GraphQl sample

Commands to test:

`sam build`

`sam local start-api`

`curl -X POST http://localhost:3000/graphql -d '{"query":"query petDetails {\n  petById(id: \"pet-1\") {\n      id\n      name\n      breed\n      owner {\n          id\n          firstName\n          lastName\n      }\n  }\n}","operationName":"petDetails"}' -H "Content-Type: application/json"
`

Test results:

`{"data":{"petById":{"id":"pet-1","name":"Alpha","breed":"Bulldog","owner":{"id":"owner-1","firstName":"Joshua","lastName":"Bloch"}}}}`

By submitting this pull request

- [ ] I confirm that my contribution is made under the terms of the Apache 2.0 license.
- [ ] I confirm that I've made a best effort attempt to update all relevant documentation.

### CODE REVIEW

**Comment by jonife** [comment]
Hey Fredric do mind putting some description on this pr for context?

**Comment by mbfreder** [comment]
Hey @jonife, I just linked the relevant issue and also added instructions to test. Thanks for the feedback.

File: samples/springboot3/graphql-pet-store/template.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    EndpointConfiguration: REGIONAL
+
+Resources:
+  PetStoreFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: com.amazonaws.serverless.sample.springboot3.StreamLambdaHandler::handleRequest
+      Runtime: java21
+      CodeUri: .
+      MemorySize: 1512
```
Comment (valerena) [inline samples/springboot3/graphql-pet-store/template.yml:17]: I saw the other springboot pet-store example already has `1512`, but it's a weird number. Normally the memory comes in like powers of 2 (`512` is in some other examples). I feel like someone just added a `1` when they wanted more memory to end up with the `1512`. But it looks better to use `1024` if that's enough memory (that's what the `alt-petstore` uses).

Comment (mbfreder) [reply samples/springboot3/graphql-pet-store/template.yml]: Updated.

File: samples/springboot3/graphql-pet-store/src/main/java/com/amazonaws/serverless/sample/springboot3/Application.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.web.servlet.HandlerAdapter;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+import org.springframework.web.servlet.HandlerMapping;
+import org.springframework.web.servlet.ModelAndView;
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply samples/springboot3/graphql-pet-store/src/main/java/com/amazonaws/serverless/sample/springboot3/Application.java]: Done, Thanks.

File: samples/springboot3/graphql-pet-store/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Serverless Spring Boot 3 and Spring-graphQl example
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply samples/springboot3/graphql-pet-store/README.md]: Ok, makes sense. I went with the first option.

**Comment by roger-zhangg** [comment]
One suggestion: I think the official name is `GraphQL`

**Pull request merged**